File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int number, line, hash, dot;
    //request input
    do
    {
        number = get_int("What height do you want the pyramid?\n");
    }
    //ensure keeps to size requirements
    while (number < 1 || number > 8);
    //make pyramid
    for (line = 0; line < number; line++)
    {
        for (dot = 0; dot < number - line - 1; dot++)
        {
            printf(" ");
        }
        for (hash = 0; hash <= line; hash++)
        {
            printf("#");
        }
        printf("\n");
    }
}

File: mario.py
from cs50 import get_int

#get positive int 1-8
#ask agin if not pos or over 8
while True:
    answer = get_int("Height:\n")
    if answer > 0 and answer < 9:
        break

#make and allign pyramid
for row in range(1, answer +1):
    print(" " * (answer - row) + "#" * row)


